flatsurf/geometry/categories/cone_surfaces.py

```diff
@@ -0,0 +1,444 @@
+r"""
+The category of cone surfaces.
+
+A cone surface is a surface that can be built by gluing Euclidean polygons
+along their edges such that the matrix describing monodromy along a closed path
+is an isometry; that matrix is given by multiplying the individual matrices
+that describe how to transition between pairs of glued edges, see
+:meth:`~.similarity_surfaces.SimilaritySurfaces.Oriented.ParentMethods.edge_matrix`.
+
+In sage-flatsurf, we restrict cone surfaces slightly by requiring that a cone
+surface is given by polygons such that each edge matrix is an isometry.
+
```

Comment:
Just a comment: if we have a "mathematical" cone surface with "wrong" edge gluing then one can apply some GL(2,R) matrix to the polygons to make it a "flatsurf" cone surface. This is worth being implemented (probably not necessarily in this PR). It would be nice to think about a user interface for "fix the gluings and find the best category".

Reply:
Yes, that's similar to the discussion about rationality. It would be nice to simplify surfaces in that way.